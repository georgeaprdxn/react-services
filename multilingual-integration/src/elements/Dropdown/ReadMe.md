Dropdown

```jsx inside Markdown
import mockData from './__test__/Dropdown.mock.json'
const [selectedOption, setSelected] = React.useState(false)
const changeHandler = selectedOption => setSelected(selectedOption)
;<Dropdown
	options={mockData.options}
	selectedOption={selectedOption}
	handleChange={selectedOption => changeHandler(selectedOption)}
/>
```
