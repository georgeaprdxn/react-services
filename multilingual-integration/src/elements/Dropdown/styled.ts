import styled from 'styled-components'

const SelectContainer = styled.div`
	width: 20%;
`

export { SelectContainer }
