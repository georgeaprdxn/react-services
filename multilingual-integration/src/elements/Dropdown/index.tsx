import React, { FunctionComponent } from 'react'
import { default as ReactSelect } from 'react-select'

import { SelectContainer } from './styled'

type Props = {
	/**
	 * handles change event
	 */
	handleChange: (selectedOption: any) => void
	/**
	 * select list option with the key value pair of label and value
	 */
	options: {
		label: string
		value: string
	}[]
	/**
	 *  selected option from the dropdown
	 */
	selectedOption: {
		label: string
		value: string
	}
	/**
	 *  to select multiple values
	 */
	isMulti?: boolean
}

const Dropdown: FunctionComponent<Props> = ({
	selectedOption,
	handleChange,
	options,
	isMulti,
}) => {
	return (
		<SelectContainer className="custom-select-container">
			<ReactSelect
				value={selectedOption}
				onChange={handleChange}
				options={options}
				className="custom-select"
				classNamePrefix="custom-select"
				isMulti={isMulti}
			/>
		</SelectContainer>
	)
}

export default Dropdown
