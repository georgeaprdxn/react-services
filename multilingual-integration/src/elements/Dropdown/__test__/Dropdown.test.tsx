import React from 'react'
import { shallow } from 'enzyme'

import Dropdown from '../index'
import mockDropdownData from './Dropdown.mock.json'

describe('Dropdown element tests', () => {
	let dropdownComp: any = null
	const handleChangeStub = jest.fn()
	beforeEach(() => {
		dropdownComp = shallow(
			<Dropdown
				options={mockDropdownData.options}
				selectedOption={mockDropdownData.selectedOption}
				isMulti={mockDropdownData.isMulti}
				handleChange={handleChangeStub}
			/>
		)
	})

	it('validates all props', () => {
		expect(dropdownComp.find('.custom-select').prop('options')).toEqual(
			mockDropdownData.options
		)
		expect(dropdownComp.find('.custom-select').prop('value')).toEqual(
			mockDropdownData.selectedOption
		)
		expect(dropdownComp.find('.custom-select').prop('isMulti')).toEqual(
			mockDropdownData.isMulti
		)
	})

	it('checks the DOM events', () => {
		dropdownComp
			.find(`.custom-select`)
			.simulate('change', { value: mockDropdownData.selectedOption })

		expect(handleChangeStub).toHaveBeenCalledTimes(1)
	})

	it('matches the snaphot', () => {
		expect(dropdownComp).toMatchSnapshot()
	})
})
