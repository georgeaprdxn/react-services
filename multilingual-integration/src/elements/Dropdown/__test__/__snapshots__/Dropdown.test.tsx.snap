// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown element tests matches the snaphot 1`] = `
<styled.div>
  <StateManager
    className="custom-select"
    classNamePrefix="custom-select"
    defaultInputValue=""
    defaultMenuIsOpen={false}
    defaultValue={null}
    isMulti={false}
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "value": Object {
                "label": "Orange",
                "value": "orange",
              },
            },
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
    options={
      Array [
        Object {
          "label": "Apple",
          "value": "apple",
        },
        Object {
          "label": "Mango",
          "value": "mango",
        },
        Object {
          "label": "Pineapple",
          "value": "pineapple",
        },
        Object {
          "label": "Orange",
          "value": "orange",
        },
      ]
    }
    value={
      Object {
        "label": "Orange",
        "value": "orange",
      }
    }
  />
</styled.div>
`;
